vortex-array/src/builders/mod.rs

```diff
@@ -88,7 +88,7 @@ pub fn builder_with_capacity(dtype: &DType, capacity: usize) -> Box<dyn ArrayBui
 pub trait ArrayBuilderExt: ArrayBuilder {
     /// A generic function to append a scalar to the builder.
     fn append_scalar(&mut self, scalar: &Scalar) -> VortexResult<()> {
-        if !scalar.dtype().eq_ignore_nullability(self.dtype()) {
+        if scalar.dtype() != self.dtype() {
```

Comment:
I don't mind being strict here, but we obviously could potentially choose to accept the scalar in the additional case where `scalar.dtype().eq_ignore_nullability(self.dtype()) && !scalar.is_null()`...

Reply:
I think for builders it's reasonable to be strict right?

Reply:
+1